### PR TITLE
feat: migrate docs site to `astro/starlight`

### LINKED ISSUE #221: feat: Add search to docs site
**Description**

Finding info on the docs site is kinda hard, having a search button would be nice.

**Requirements**

- [ ] Add Search

**Additional Context**

None

### PR DESCRIPTION
**Pull request by felangel**
## Status

**READY**

## Description

- feat: migrate docs site to astro/starlight
  - closes #221

### CODE REVIEW

File: src/content/docs/code-push/patch.mdx [hunk 1]
```diff
@@ -21,7 +20,7 @@ shorebird patch android
 
 </TabItem>
 
-<TabItem value="patch-command-ios" label="iOS">
+<TabItem value="patch-command-ios" label="iOS (beta)">
```
Comment (eseidel) [inline src/content/docs/code-push/patch.mdx:23]: Oops.

File: src/content/docs/code-push/patch.mdx [hunk 2]
```diff
@@ -67,18 +66,10 @@ Would you like to continue? (y/N) Yes
...
 ```
 
-By default, this uses the release version that the app is currently on. If you
-want to patch a different release version, you can use the `--release-version`
-option. For example:
-
-```
-shorebird patch android --release-version 0.1.0+1
-```
-
```
Comment (eseidel) [inline src/content/docs/code-push/patch.mdx:77]: Why removed?

File: src/content/docs/code-push/patch.mdx [hunk 3]
```diff
@@ -67,18 +66,10 @@ Would you like to continue? (y/N) Yes
...
 If your application supports flavors or multiple release targets, you can specify the flavor and target using the `--flavor` and `--target` options:
 
 ```
-shorebird patch [android|ios] --target lib/main_development.dart --flavor development
+shorebird patch --target ./lib/main_development.dart --flavor development
```
Comment (eseidel) [inline src/content/docs/code-push/patch.mdx:72]: ?

File: src/content/docs/code-push/release.mdx [hunk 4]
```diff
@@ -1,33 +1,42 @@
...
-
-## Publish a Release
+import { Tabs, TabItem } from '@astrojs/starlight/components';
 
 In order to start pushing updates, you will need to create a release.
 
-Creating a release builds and submits your app to Shorebird. Shorebird saves the
-compiled Dart code from your application in order to make updates smaller in
-size.
-
```
Comment (eseidel) [inline src/content/docs/code-push/release.mdx:19]: ?

**Review by erickzanardo — APPROVED** [review]

File: src/content/docs/code-push/release.mdx [hunk 5]
```diff
@@ -55,51 +64,11 @@ See the following link for more information:
 https://support.google.com/googleplay/android-developer/answer/9859152?hl=en
 ```
 
-If your application supports flavors or multiple release targets, you can specify the flavor and target using the `--flavor` and `--target` options:
-
-```bash
-shorebird release android --target ./lib/main_development.dart --flavor development
-```
-
-:::info
```
Comment (eseidel) [inline src/content/docs/code-push/release.mdx:64]: ?

File: src/content/docs/guides/release/ios.mdx [hunk 6]
```diff
@@ -1,21 +1,20 @@
...
 ---
 
-# Releasing to the Apple App Store
-
 This guide walks through releasing a code push app to the Apple App Store and applying a patch to that release.
 
 The app we will be releasing in this guide is Shorebird Clock, our demo code push app. ([source](https://github.com/shorebirdtech/time_shift/))
 
-## Prerequisites
+## Prequisites
```
Comment (eseidel) [inline src/content/docs/guides/release/ios.mdx:13]: ?

File: src/content/docs/status.mdx [hunk 7]
```diff
@@ -1,18 +1,15 @@
...
-## Stable
-
 Shorebird is available on Android and iOS, with ~1000 apps using Shorebird on
 the App Store and Play Store.
 
 ## Link Percentage (iOS)
 
 As part of [complying with the the App Store
-guidelines](faq#does-shorebird-comply-with-app-store-guidelines), Shorebird uses
+guidelines](/faq#does-shorebird-comply-with-play-store-guidelines), Shorebird uses
```
Comment (eseidel) [inline src/content/docs/status.mdx:12]: Wrong one?

File: src/content/docs/troubleshooting.mdx [hunk 8]
```diff
@@ -191,7 +188,7 @@ native code of your app, and are not patchable by Shorebird. If you see this
 warning, you should be very careful about publishing your patch, as it may
 cause your app to crash when the Dart code tries to call into native code
 which operates differently than expected. See
-[the next section](#unexpected-native-changes).
+[the next section](#i-see-a-the-release-artifact-contains-native-changes-warning-when-running-shorebird-patch-even-though-i-havent-changed-swiftobjective-ckotlinjava-code-unexpected-native-changes).
```
Comment (eseidel) [inline src/content/docs/troubleshooting.mdx:191]: Wrong one?

**Review by eseidel — APPROVED** [review]

File: src/content/docs/status.mdx (hunk 7 above)
Comment (felangel) [reply src/content/docs/status.mdx:12]: no that's the first of two store links (the only one was removed).

File: src/content/docs/troubleshooting.mdx (hunk 8 above)
Comment (felangel) [reply src/content/docs/troubleshooting.mdx:191]: The old one was removed so I think this is correct (or we can remove the link).

File: src/content/docs/guides/release/ios.mdx [hunk 9]
```diff
@@ -34,7 +33,7 @@ To build an iOS app for distribution, we need to specify a development team in X
 
 Navigate to your app on the [Shorebird console](https://console.shorebird.dev/)
 to see the current set of releases. For our app, we see that the latest release
-version is `1.0.3+1`, so the version of our next release will be `1.0.4+1`.
+version is `1.0.3.+1`, so the version of our next release will be `1.0.4+1`.
```
Comment (eseidel) [inline src/content/docs/guides/release/ios.mdx:36]: Extra .

**Review by eseidel — APPROVED** [review]

**Pull request merged**